types/koa-generic-session/koa-generic-session-tests.ts

```diff
@@ -32,4 +32,16 @@ app.use(session({
     beforeSave: (ctx: Koa.Context, session: Session) => {}
 }));
 
+app.use((context: Koa.Context) => {
+    if (! context.session) {
+        return;
+    }
+
+    context.regenerateSession();
+    context.sessionSave = true;
+    context.session.cookie;
+    context.session.key = 'value';
```

Comment:
You can't access the session keys like this when you define it with `[key: string]: any;`. 

`context.session['key'] = 'value';` should work.